src/session.rs

```diff
@@ -383,4 +430,48 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn can_read_session_with_duplicate_keys() -> Result<()> {
+        let mut path_to_session = std::env::temp_dir();
+        let file_name = random_string();
+        path_to_session.push(file_name);
+        fs::write(
```

Comment:
Should we remove file operations and focus on just the permutations of session content that xh can load? Our integration tests should cover the rest (I think).

Reply:
Yeah, I don't think this is the right place to test the I/O.

Maybe the `load_session` constructor should be split into two layers so you can easily feed it a string.